transporters/src/main/java/com/nextdoor/bender/ipc/splunk/SplunkTransportConfig.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * Copyright 2017 Nextdoor.com, Inc
+ *
+ */
+
+package com.nextdoor.bender.ipc.splunk;
+
+import java.io.UnsupportedEncodingException;
+
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.kjetland.jackson.jsonSchema.annotations.JsonSchemaDefault;
+import com.kjetland.jackson.jsonSchema.annotations.JsonSchemaDescription;
+import com.nextdoor.bender.ipc.TransportConfig;
+import com.nextdoor.bender.utils.Passwords;
+
+@JsonTypeName("Splunk")
+@JsonSchemaDescription("Writes events to a Splunk HEC endpoint.")
+public class SplunkTransportConfig extends TransportConfig {
+
+  @JsonSchemaDescription("Splunk HTTP endpoint hostname.")
+  @JsonProperty(required = true)
+  private String hostname = null;
+
+  @JsonSchemaDescription("Splunk HTTP endpoint port.")
+  @JsonSchemaDefault(value = "443")
+  @JsonProperty(required = false)
+  @Min(1)
+  @Max(65535)
+  private Integer port = 443;
+
+  @JsonSchemaDescription("Splunk auth token.")
+  @JsonProperty(required = true)
+  private String authToken = null;
+
+  @JsonSchemaDescription("Splunk data index.")
+  @JsonProperty(required = true)
```

Comment:
In theory, this shouldn't be requires - as the HEC Token is supposed to default to the correct configuration stanza in the `inputs.conf` file.